apps/frontend/src/pages/Charities.tsx

```diff
@@ -0,0 +1,258 @@
+import { useState, useEffect, ChangeEvent } from "react";
+import {
+  Typography,
+  Grid,
+  Switch,
+  FormControlLabel,
+  Box,
+  Avatar,
+  Paper,
+} from "@mui/material";
+import CharityCard from "../components/CharityCard";
+import SearchBar from "../components/searchBar";
+import { colors } from "../styles/colors";
+import _axios_instance from "../_axios_instance";
+import Spinner from "../components/Spinner";
+
+interface Organization {
+  name: string;
+  logoUrl: string;
+  donated: number;
+  receiving: boolean;
+}
+
+interface Charity {
+  id: string;
+  name: string;
+  description: string;
+  startDate: string;
+  endDate: string;
+  imageUrl: string;
+  organizations: Organization[];
+  funds: number;
+  listingsCount: number;
+}
+
+const Charities = () => {
+  const currentDate = new Date();
+  const [showPastEvents, setShowPastEvents] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState<string>("");
+  const [charities, setCharities] = useState<Charity[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await _axios_instance.get("/charities");
+        setCharities(response.data);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching charities:", error);
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, []);
+
+  const currentCharity = Array.isArray(charities)
+    ? charities.find((charity) => new Date(charity.endDate) >= currentDate)
+    : undefined;
+
+  const pastCharities = Array.isArray(charities)
+    ? charities.filter((charity) => new Date(charity.endDate) < currentDate)
+    : [];
+
+  useEffect(() => {
+    if (currentCharity) {
+      const endTime = new Date(currentCharity.endDate).getTime();
+      const interval = setInterval(() => {
+        const now = new Date().getTime();
+        const distance = endTime - now;
+
+        if (distance < 0) {
+          clearInterval(interval);
+          setTimeRemaining("This charity event has ended.");
+        } else {
+          const days = Math.floor(distance / (1000 * 60 * 60 * 24));
+          const hours = Math.floor(
+            (distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60),
+          );
+          const minutes = Math.floor(
+            (distance % (1000 * 60 * 60)) / (1000 * 60),
+          );
+          const seconds = Math.floor((distance % (1000 * 60)) / 1000);
+          setTimeRemaining(`${days}d ${hours}h ${minutes}m ${seconds}s`);
```

Comment:
Again, looking into JS date to string formatting might make this a lot cleaner

Reply:
Since there's math involved in calculating the live countdown number, I don't think there's much simplification I can do 🤔